numbers/src/test/java/org/dicio/numbers/lang/fr/PronounceNumberTest.java

```diff
@@ -0,0 +1,240 @@
+        assertEquals("mille-cinq-cent-soixante-septième", pf.pronounceNumber(1567).ordinal(T).shortScale(T).get());
+        assertEquals("mille-cinq-cent-soixante-septième", pf.pronounceNumber(1567).ordinal(T).shortScale(F).get());
+
+        // big numbers
+        assertEquals("dix-huit-millionième", pf.pronounceNumber(18000000).ordinal(T).get());
+        assertEquals("dix-huit-mille-centième", pf.pronounceNumber(18000100).ordinal(T).get());
+        assertEquals("cent-vingt-sept-milliardième", pf.pronounceNumber(127000000000.0).ordinal(T).shortScale(T).get());
+        assertEquals("deux-cent-un-millionième", pf.pronounceNumber(201000000000.0).ordinal(T).shortScale(F).get());
+        //TODO: Check this for french correctness as well as short/long scale issues (billion/billard)
+        assertEquals("neuf-cent-treize-milliard-quatre-vingt-million-six-cent-mille-soixante-mille-cent-soixante-quatrième", pf.pronounceNumber(913080600064.0).ordinal(T).shortScale(T).get());
+        assertEquals("neuf-cent-treize-mille-quatre-vingt-million-six-cent-mille-soixante-mille-soixante-quatrième", pf.pronounceNumber(913080600064.0).ordinal(T).shortScale(F).get());
+        assertEquals("trilliard-deux-millionième", pf.pronounceNumber(1000002000000.0).ordinal(T).shortScale(T).get());
+        assertEquals("millard-deux-millionième", pf.pronounceNumber(1000002000000.0).ordinal(T).shortScale(F).get());
+        assertEquals("quatre-triliard-un-millionième", pf.pronounceNumber(4000001000000.0).ordinal(T).shortScale(T).get());
+        assertEquals("quatre-milliard-un-millionième", pf.pronounceNumber(4000001000000.0).ordinal(T).shortScale(F).get());
+
+        // decimal numbers and scientific notation: the behaviour should be the same as with ordinal=F
+        assertEquals("deux virgule soixante-dix-huit", pf.pronounceNumber(2.78).ordinal(T).get());
+        assertEquals("tiers", pf.pronounceNumber(2.78).places(0).ordinal(T).get());
+        assertEquals("dix-neuf", pf.pronounceNumber(19.004).ordinal(T).get());
+        assertEquals("huit cent trente-trois millions, quatre cent vingt-huit mille, quatre-vingt-douze virgule cent quatre-vingt-trois", pf.pronounceNumber(830438092.1829).places(3).ordinal(T).get());
+        assertEquals("deux virgule cinquante-quatre fois dix puissance six", pf.pronounceNumber(2.54e6).ordinal(T).scientific(T).get());
+    }
+
+    @Test
+    public void edgeCases() {
+        assertEquals("zéro", pf.pronounceNumber(0.0).get());
+        assertEquals("zéro", pf.pronounceNumber(-0.0).get());
+        assertEquals("infini", pf.pronounceNumber(Double.POSITIVE_INFINITY).get());
+        assertEquals("moins l'infini", pf.pronounceNumber(Double.NEGATIVE_INFINITY).scientific(F).get());
+        assertEquals("moins l''infini", pf.pronounceNumber(Double.NEGATIVE_INFINITY).scientific(T).get());
+        assertEquals("Non défini", pf.pronounceNumber(Double.NaN).get());
```

Comment:
"non défini" or the litteral translation "pas un nombre" ?

Reply:
"Non défini", as in 10 divided by 0.